### PR TITLE
add stub_with/2 for stubbing whole module

### PR DESCRIPTION
**Pull request by calvin-kargo**
As per https://github.com/edgurgel/mimic/issues/30

Adding `stubs_with/2` function which could mock a module with another module

### CODE REVIEW

**Comment by edgurgel** [comment]
Thanks for sending this PR!  ❤️ 

I gave a quick look and it looks pretty good 👍 but at the moment pretty busy! I hope that I have some time next week to review it properly

File: lib/mimic.ex [hunk 1]
```diff
@@ -162,6 +162,42 @@ defmodule Mimic do
...
+      defmodule InverseCalculator do
+        def add(a, b), do: a - b
+      end
+
+      iex> Mimic.stubs_with(Calculator, InverseCalculator)
+      ...> Calculator.add(2, 4)
+      ...> -2
+
+  """
+  @spec stub(module()) :: module()
```
Comment (edgurgel) [inline lib/mimic.ex:190]: Should we call it `stub_with` ?

Should this be `@spec stub_with` ?

Comment (calvin-kargo) [reply lib/mimic.ex:190]: ah, definitely, let me change that

Comment (calvin-kargo) [reply lib/mimic.ex]: changed @edgurgel

**Pull request merged**